cli/src/stake.rs

```diff
@@ -1980,17 +1981,28 @@ pub fn process_split_stake(
 
     let rent_exempt_reserve = if !sign_only {
         if let Ok(stake_account) = rpc_client.get_account(&split_stake_account_address) {
-            let err_msg = if stake_account.owner == stake::program::id() {
-                format!("Stake account {split_stake_account_address} already exists")
-            } else {
-                format!(
-                    "Account {split_stake_account_address} already exists and is not a stake \
-                     account"
-                )
+            if stake_account.owner == stake::program::id() {
+                return Err(CliError::BadParameter(format!(
+                    "Stake account {split_stake_account_address} already exists"
+                ))
+                .into());
+            }
+            if stake_account.owner == system_program::id() {
+                if stake_account.data.len() > 0 {
+                    return Err(CliError::BadParameter(format!(
+                        "Account {split_stake_account_address} has data and cannot be used to split stake"
+                    ))
+                    .into());
+                }
+
+                if stake_account.lamports == 0 {
```

Comment:
this won't happen via rpc. it's by definition an unallocated account

Reply:
But the issue's proposed system says the destination split stake account can be a system account without any data and some lamports  so that it can be allocated some data and it's owner is assigned to stake program later

Reply:
ty ser

Reply:
Not a ser